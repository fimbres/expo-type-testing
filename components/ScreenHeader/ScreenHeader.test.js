import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react-native';

import { ScreenHeader } from '.';

describe('<ScreenHeader />', () => {
    it('render both texts correctly', () => {
        render(
            <ScreenHeader
                title='This is a title'
                subtitle='This is a subtitle'
            />
        );

        const title = screen.getByText('This is a title');
        const subtitle = screen.getByText('This is a subtitle');

        expect(title).toBeDefined();
        expect(subtitle).toBeDefined();
    });

    it('render correctly', () => {
        const tree = renderer
            .create(
                <ScreenHeader
                    title='This is a title'
                    subtitle='This is a subtitle'
                />
            ).toJSON();
        expect(tree.children.length).toBe(2);
        expect(tree).toMatchSnapshot();
    });
});
