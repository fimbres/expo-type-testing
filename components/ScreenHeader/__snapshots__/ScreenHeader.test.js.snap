// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScreenHeader /> render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 20,
        "paddingTop": 24,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#020202",
        "fontFamily": "avenir-black",
        "fontSize": 20,
        "fontWeight": "800",
        "lineHeight": 27.32,
      }
    }
  >
    This is a title
  </Text>
  <Text
    style={
      Object {
        "color": "#020202",
        "fontFamily": "avenir-book",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 21.86,
      }
    }
  >
    This is a subtitle
  </Text>
</View>
`;
